### ISSUE TITLE
Deleted files do not seem to be part of the "exported --report"

### ISSUE THREAD

**Issue by oPromessa** [RhetTbull/osxphotos#853]
**Before submitting a bug report, please ensure you are running the most recent version of osxphotos and that the bug is reproducible on the latest version**

* Done!

**Describe the bug**
* `exported --report` does not seem to be reporting delete files (as per feature #395) 

**To Reproduce**
1. What' the full command line you used with osxphotos?
##### Command line on the crontab
```bash
osxphotos export "${DSTFOLDER}" --exportdb ${FOLDER}/.osxphotos_export.db --load-config ${WORKFOLDER}/colibri.toml >> ${LOGFILE} 2>&1
```
##### TOML file being used
```toml
[export]
cleanup = true
db = "/Users/Shared/Pictures/iPhoto Shared Library.photoslibrary"
description_template = "Album:{album,}{newline}Description:{descr,}"
directory = "{album|filter(startswith 1|2)[/,.|:,.]}"
exiftool = true
exiftool_merge_keywords = true
exiftool_merge_persons = true
keyword_template = [ "{keyword}", "{label}", "{searchinfo.activity?activity:{searchinfo.activity},}", "{searchinfo.venue_type?venue_type:{searchinfo.venue_type},}", ]
no_progress = true
not_hidden = true
ramdb = true
retry = 2 
skip_bursts = true
skip_live = true
skip_original_if_edited = true
strip = true
timestamp = true
touch_file = true
update = true
```

2. What was the error output?
```bash
$ tail runOSxPhotos.cron.err
2022-12-03 03:45:10.980864 -- Exporting 88919 photos to 
/Volumes/photo-1/Family.Photos...
2022-12-03 09:44:53.879977 -- Processed: 88919 photos, exported: 3, updated: 
215, skipped: 88703, updated EXIF data: 218, missing: 0, error: 0, touched date:
244
2022-12-03 09:44:53.962077 -- Elapsed time: 5:59:43
2022-12-03 09:44:54.010461 -- Cleaning up /Volumes/photo-1/Family.Photos
2022-12-03 09:49:32.259566 -- Deleted: 4 files, 0 directories
[notice] Result: 137
[notice] --- End Sat 3 Dec 2022 10:58:18 WET
$ osxphotos  exportdb --report colibri.json 0 .osxphotos_export.db 
Wrote report to colibri.json
$ head colibri.json 
[{
    "datetime": "2022-12-03T03:45:10.980793",
    "filename": "/Volumes/photo-1/(...)/DSC03760.JPG",
    "exported": false,
    "new": false,
    "updated": false,
    "skipped": true,
    "exif_updated": false,
    "touched": false,
    "converted_to_jpeg": false,
$ grep deleted colibri.json | grep -v false
$ 
```

**Expected behavior**
* To have deleted files reported on the report filewuth the page `cleanup_deleted_file: true`.

**Screenshots**
* See above.

**Desktop (please complete the following information):**
 - OS: [e.g. which version of macOS]
ProductName:	macOS
ProductVersion:	12.6.1
BuildVersion:	21G217
 - osxphotos version (`osxphotos --version`)
osxphotos, version 0.54.4

**Additional context**
* Curiously enough `osxphotos` exited with return code **137** on the last run. No more details on it and also no python crash log found!.
* So I'll run with the `verbose` option tonight and deleting one file to see if I can catch any error.

**Comment by oPromessa** [comment]
Hi again. Did a test in a more controlled environment with `verbose` and could also reproduce the issue.
* In this case the deleted file is present in two folders (Albums).

```bash
$ tail local.out 
2022-12-03 11:40:26.808336 -- Processed: 10 photos, exported: 0, updated: 0, skipped: 10, updated EXIF data: 0, missing: 0, error: 0, touched date: 0
2022-12-03 11:40:26.809169 -- Elapsed time: 0:00:01
2022-12-03 11:40:26.809568 -- Cleaning up /Users/Shared/Pictures/Export/bad_pics_export_osxphotos/local
2022-12-03 11:40:26.810257 -- Deleting /Users/Shared/Pictures/Export/bad_pics_export_osxphotos/local/2006(...)/PC150216.JPG
2022-12-03 11:40:26.811221 -- Deleting /Users/Shared/Pictures/Export/bad_pics_export_osxphotos/local/Bad Photos/PC150216.JPG
2022-12-03 11:40:26.811874 -- Deleting empty directory /Users/Shared/Pictures/Export/bad_pics_export_osxphotos/local/2006(...)
2022-12-03 11:40:26.812565 -- Deleting empty directory /Users/Shared/Pictures/Export/bad_pics_export_osxphotos/local/Bad Photos
2022-12-03 11:40:26.812966 -- Deleted: 2 files, 2 directories
2022-12-03 11:40:26.813364 -- Wrote export report to local.csv
2022-12-03 11:40:26.813726 -- Writing export database changes back to /Users/Shared/Pictures/Export/bad_pics_export_osxphotos/local/.osxphotos_export.db
$ osxphotos  exportdb --report local.json 0 local/.osxphotos_export.db 
Wrote report to local.json
$ grep cleanup local.json 
    "cleanup_deleted_file": false,
    "cleanup_deleted_directory": false,
    "cleanup_deleted_file": false,
    "cleanup_deleted_directory": false,
    "cleanup_deleted_file": false,
    "cleanup_deleted_directory": false,
    "cleanup_deleted_file": false,
    "cleanup_deleted_directory": false,
    "cleanup_deleted_file": false,
    "cleanup_deleted_directory": false,
    "cleanup_deleted_file": false,
    "cleanup_deleted_directory": false,
    "cleanup_deleted_file": false,
    "cleanup_deleted_directory": false,
```

**Comment by oPromessa** [comment]
An update after some more tests:
* the file generated with the `--report` option in *export* command: `osxphotos export (...)   --report "local.csv"` **does** include the deleted file names from the **run**

Possibly the deleted entries are not recorded on `.osxphotos_export.db`and thus the `exportdb --report` is not able to recoup them... which makes sense.

So this is possibly not a **bug**!!!

**Comment by RhetTbull** [comment]
I figured this out.  The export results are written to the database before the cleanup happens thus they deleted files are not available to `exportdb --report`.  This is an easy fix but I'm in the middle of several other projects so will get to it when I can.

Notice the call to `set_export_results` happens before the cleanup code (`if cleanup:`)

https://github.com/RhetTbull/osxphotos/blob/f8d8028631de3db697564a8f53c02ee4a613c913/osxphotos/cli/export.py#L1714-L1744

**Comment by oPromessa** [comment]
Great. No worries. Take your time. Thanks again for all the hard-work on this great tool.

**Comment by RhetTbull** [comment]
Fixed in [v0.55.3](https://github.com/RhetTbull/osxphotos/releases/tag/v0.55.3)
